### PR TITLE
finally fixed that annoying layout issue

### LINKED ISSUE #334: 404 page doesn't look good
### Before you start...

- [X] Have you refreshed the page? You might be using an old version of the site.
- [X] Have you checked for an existing issue on this topic? If there is one, post a comment on it instead.

### What browsers are you seeing the problem on?

Microsoft Edge, Firefox Beta/Dev/Nightly

### Description

The error window is't placed in proper place, so it makes harder to read messages.

### Steps To Reproduce

1. Go to https://files.community/404 (an example)

### Expected behavior

I think the error window should overlap neither navbar nor page-section-inner

### Relevant Assets

image

### PR DESCRIPTION
**Pull request by TheOnlyTails**
## Description

for a while, we've had an issue with layout stretching and flexbox. this PR finally fixes that!
For example:
- The error page now displays correctly
- Documentation pages no longer have an inner scrolling page, but instead grow to fill up more vertical space.

## Motivation and Context

Closes #334 

## Screenshots (if appropriate):

image

### CODE REVIEW

File: src/styles/_global.scss [hunk 1]
```diff
@@ -108,8 +108,8 @@ textarea {
 // Outer elements -- <html> and <body>
 html,
 body {
-	inline-size: 100%;
-	block-size: 100%;
+	block-size: 100vh;
+	block-size: 100dvh;
```
Comment (chingucoding) [inline src/styles/_global.scss:112]: Double block size?

File: src/hooks.server.ts [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import { locales } from "$i18n";
 import type { Handle } from "@sveltejs/kit";
-import { unwrapFunctionStore, locale } from "svelte-i18n";
+import { locale, unwrapFunctionStore } from "svelte-i18n";
```
Comment (chingucoding) [inline src/hooks.server.ts:3]: Unrelated.

**Review by chingucoding — COMMENTED** [review]
Those are a lot of changes for fixing the error page. Are they all needed?

File: src/styles/_global.scss (hunk 1 above)
Comment (TheOnlyTails) [reply src/styles/_global.scss:112]: this is intended: on devices where dynamic viewport height is supported we want to use that, but if it's not supported that rule will be ignored and regular viewport height will be used instead.

File: src/hooks.server.ts (hunk 2 above)
Comment (TheOnlyTails) [reply src/hooks.server.ts:3]: i think the auto-formatter made this change, i can revert it if need be.

**Comment by TheOnlyTails** [comment]
This change isn't just to fix the error page, it's intended to fix multiple issues with the page layout system we had in place till now.

File: src/styles/_global.scss (hunk 1 above)
Comment (chingucoding) [reply src/styles/_global.scss:112]: Only IE and Opera don't support this but I'm a bit surprised Svelte or Sass don't take care of that.

Comment (TheOnlyTails) [reply src/styles/_global.scss:112]: It's not just which browser, but what version. All of these are relatively new versions of the evergreen browsers, and it's only at 75% global support.
image

Comment (TheOnlyTails) [reply src/styles/_global.scss:112]: Also, neither CSS nor JS frameworks like Sass and Svelte should interfere in your styles. That's what postcss is for.

Comment (chingucoding) [reply src/styles/_global.scss:112]: Oh didn't thought it was that recent. Still surprised we need to think of this ourselfes given tools exist that handle fallbacks like that.

**Comment by chingucoding** [comment]
Not sure why that is happening but it seems when the page is narrow, the documentation TreeView is not expanding anymore:

Screenshot of navigation not showing all items

Can you repro this?

File: src/styles/_global.scss (hunk 1 above)
Comment (chingucoding) [reply src/styles/_global.scss:112]: I would honestly expect my CSS transform to also provide a fallback for stuff like that. IIRC Webpack also does this though I haven't used that in ages.

**Comment by TheOnlyTails** [comment]
This is because I hadn't merged #343 yet, it should be fixed now.

**Review by chingucoding — APPROVED** [review]
